web/modules/custom/mymodule/src/Plugin/Block/Stockblock.php

```diff
@@ -0,0 +1,139 @@
+
+    }
+    return [
+      '#type' => 'inline_template',
+      '#template' => "{% trans %}<strong> Stock Information: </strong><br/>{% endtrans %} {{stockname | raw }} {{stockdesc | raw }} {{pricedetails | raw }}",
+      '#context' => [
+        'stockname' => '<strong>Stock Name:</strong>' . $output['name'] . '<br/>',
+        'stockdesc' => '<strong>Stock Description:</strong>' . $output['description'] . '<br/>',
+        'pricedetails' => '<strong>Price Details:</strong><br/>' . $pricedetails . '<br/>',
+      ],
+    ];
+  }
+
+  /**
+   * Block form function is used to create a form.
+   *
+   * @param object $form
+   *   We will set the field in form variable.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   It will retrieve the value of form.
+   *
+   * @return array
+   *   Return the form fields.
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form = parent::blockForm($form, $form_state);
+    $config = $this->getConfiguration();
+    $form['company_symbol'] = [
+      '#type' => 'textfield',
+      '#title' => t('Company'),
+      '#default_value' => $config['company_symbol'] ?? '',
+    ];
+    $form['start_date'] = [
+      '#type' => 'textfield',
+      '#title' => t('Start date'),
+      '#default_value' => $config['start_date'] ?? '',
+    ];
+    $form['end_date'] = [
+      '#type' => 'textfield',
+      '#title' => t('End date'),
+      '#default_value' => $config['end_date'] ?? '',
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Block submit function.
+   *
+   * @param object $form
+   *   We will set the field in form variable.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   It will retrieve the value of form.
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    $this->setConfigurationValue('company_symbol', $form_state->getValue('company_symbol'));
+    $this->setConfigurationValue('start_date', $form_state->getValue('start_date'));
+    $this->setConfigurationValue('end_date', $form_state->getValue('end_date'));
+  }
+
+  /**
+   * Disable cache on this block function.
+   *
+   * @return int
+   *   Return cache is enbled or not.
+   */
+  public function getCacheMaxAge() {
+    // If you want to disable caching for this block.
+    return 0;
+  }
+
+  /**
+   * Curl function.
+   *
+   * @param string $service_url
+   *   Return string.
+   *
+   * @return objectarray
+   *   Return objectarray.
+   */
+  public function curlResponse($service_url) {
+    // $curl = curl_init($service_url);
+    // curl_setopt($curl, CURLOPT_RETURNTRANSFER, TRUE);
+    // curl_setopt($curl, CURLOPT_POST, FALSE);
+    // curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, TRUE);
+    // $curl_response = curl_exec($curl);
+    // curl_close($curl);
+    // $output = json_decode($curl_response);
+    $client = new Client();
+    $response = $client->get($service_url);
```

Comment:
@anitha-innoraft - All the API requests should be using try and cache statements i.e. proper exceptional handling. It will make sure your application will not error out if something goes wrong in the API. Let's implement that. Here is a doc for php exceptions - https://www.drupal.org/docs/develop/coding-standards/php-exceptions

Reply:
Done